testsuite/cluster-test/src/effects/generate_cpu_flamegraph.rs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+#![forbid(unsafe_code)]
+
+use crate::{effects::Action, instance::Instance};
+use anyhow::Result;
+use async_trait::async_trait;
+use slog_scope::info;
+use std::fmt;
+
+pub struct GenerateCpuFlamegraph {
+    instance: Instance,
+    duration_secs: usize,
+}
+
+impl GenerateCpuFlamegraph {
+    pub fn new(instance: Instance, duration_secs: usize) -> Self {
+        Self {
+            instance,
+            duration_secs,
+        }
+    }
+}
+
+#[async_trait]
+impl Action for GenerateCpuFlamegraph {
+    async fn apply(&self) -> Result<()> {
+        info!("GenerateCpuFlamegraph {}", self.instance);
+        let cmd = format!(
+            r#"
+                set -e;
+                rm -rf /tmp/perf-data;
+                mkdir /tmp/perf-data;
+                cd /tmp/perf-data;
+                sudo perf record -F 99 -p $(ps aux | grep libra-node | grep -v grep | awk '{{print $2}}') --output=perf.data --call-graph dwarf -- sleep {};
+                sudo perf script --input=perf.data | sudo /usr/local/etc/FlameGraph/stackcollapse-perf.pl > out.perf-folded;
+                sudo /usr/local/etc/FlameGraph/flamegraph.pl out.perf-folded > perf-kernel.svg;
+                "#,
+            self.duration_secs
+        );
+        self.instance.run_cmd(vec![cmd.as_str()]).await?;
+        self.instance
+            .scp("/tmp/perf-data/perf-kernel.svg", "perf-kernel.svg")
```

Comment:
Does it make sense to prefix name somehow, so that one can run multiple `GenerateCpuFlamegraph` in parallel on different nodes, if needed?

Reply:
We can probably change it when we have parallel runs.